TUM Campus App/CafeteriaManager.swift

```diff
@@ -54,7 +55,7 @@ final class CafeteriaManager: CachedManager, SingleItemCachedManager, CardManage
         }
     }
     
-    func fetch(maxCache: CacheTime) -> Response<[Cafeteria]> {
+    func performRequest(maxCache: CacheTime) -> Response<[Cafeteria]> {
```

Comment:
Why call it performRequest and not fetch? Every other *Manager implements fetch and the name change is a bit inconsistent. And it seems to do the same thing (correct me if I'm wrong).

Reply:
Yeah. I wanted to call it `fetch`, but then we are in the ambiguous naming territory because it will be called exactly the same.

We want to make sure that `fetch(maxCache)` still checks the in-memory cache first.
I'm open to ideas ;)

Reply:
Maybe `fetchFromSource` (since it reflects the relationship with `fetch` better) but I don't think that solves the ambiguity problem.

Reply:
Maybe. The problem simply rises for using method composition to hide away abstractions. The second you have two default composited implementations the compiler will scream in agony for having names them identically.

A different approach would be having the fetch function performed by an private implementation of Cached Manager and using delegation. As in:

```swift
public class CafeteriaManager: MemoryCachedManager {

    private struct InternalManager: CachedManager { func fetch(maxCache: CacheTime) {} }

    // TODO: delegate fetching to internal manager on cache miss

}
```

Reply:
too complicated.

Reply:
What about wrapping it in a class instead of a protocol. Then the user will have more info on how caching works but I can live with that.: 

```swift
class MemoryCached<Manager: CachedManager>: CachedManager {



}
```